elide-dbmanager/elide-dbmanager-hibernate5/src/main/java/com/yahoo/elide/dbmanagers/hibernate5/HibernateManager.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.dbmanagers.hibernate5;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import com.yahoo.elide.core.DatabaseManager;
+import com.yahoo.elide.core.DatabaseTransaction;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.FilterScope;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.exceptions.TransactionException;
+import com.yahoo.elide.security.Check;
+import com.yahoo.elide.security.CriteriaCheck;
+import com.yahoo.elide.security.User;
+
+import org.hibernate.EntityMode;
+import org.hibernate.Hibernate;
+import org.hibernate.HibernateException;
+import org.hibernate.ObjectNotFoundException;
+import org.hibernate.ScrollMode;
+import org.hibernate.ScrollableResults;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.metadata.ClassMetadata;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import lombok.NonNull;
+import org.hibernate.resource.transaction.spi.TransactionStatus;
+
+/**
+ * Hibernate interface library
+ */
+public class HibernateManager extends DatabaseManager {
+
+    /**
+     * Wraps ScrollableResult as Iterator
+     * @param <T> type of return object
+     */
+    public static class ScrollableIterator<T> implements Iterable<T>, Iterator<T> {
+
+        final private ScrollableResults scroll;
+        private boolean inUse = false;
+        private boolean hasNext;
+
+        public ScrollableIterator(ScrollableResults scroll) {
+            this.scroll = scroll;
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            if (inUse) {
```

Comment:
Yeah. This file is copied and pasted from Hibernate3. I just modified it to work with Hibernate5.